packages/nuxt-meta-pixel/src/module.ts

```diff
@@ -13,20 +13,31 @@ export default defineNuxtModule<ModuleOptions>({
     name: 'nuxt-meta-pixel',
     configKey: 'metapixel'
   },
+  defaults: {
+    enabled: true,
+  },
   setup (options, nuxt) {
     const resolver = createResolver(import.meta.url)
     
     nuxt.options.runtimeConfig.public.metapixel = defu(
       nuxt.options.runtimeConfig.public.metapixel,
       options
     )
+    
+    if(!nuxt.options.runtimeConfig.public.metapixel.enabled) {
+      return
+    }
 
     // minimatch > brace-expansion
     // Ensure we transform these cjs dependencies, remove as they get converted to ejs
     nuxt.options.vite.optimizeDeps ||= {}
     nuxt.options.vite.optimizeDeps.include ||= []
     nuxt.options.vite.optimizeDeps.include.push('brace-expansion')
 
-    addPlugin(resolver.resolve('./runtime/plugin.client'))
+
+    addPlugin({
+      src: resolver.resolve('./runtime/plugin.client'),
+      mode: 'client'
+    })
```

Comment:
Is there a reason to specify `mode: 'client'` ?
Following https://nuxt.com/docs/guide/directory-structure/plugins the naming `plugin.client` already activates the plugin for the client only.

Reply:
Yes you're right. This is unnecessary.